components/support/nimbus-fml/src/editing/values_finder.rs

```diff
@@ -0,0 +1,161 @@
+            feature_value,
+        }
+    }
+
+    pub(crate) fn all(&self, type_ref: &TypeRef) -> BTreeSet<String> {
+        match type_ref {
+            TypeRef::StringAlias(_) => self.get_string_alias_values(type_ref),
+            TypeRef::Enum(type_name) => self.get_enum_values(type_name),
+            _ => Default::default(),
+        }
+    }
+}
+
+impl<'a> ValuesFinder<'a> {
+    fn get_enum_values(&self, type_name: &str) -> BTreeSet<String> {
+        if let Some(def) = self.enum_defs.get(type_name) {
+            def.variants.iter().map(|v| v.name()).collect()
+        } else {
+            Default::default()
+        }
+    }
+
+    fn get_string_alias_values(&self, alias_type: &TypeRef) -> BTreeSet<String> {
+        let type_name = alias_type.name().unwrap();
+        let prop = self.string_aliases[type_name];
+
+        let def_type = &prop.typ;
+        let def_value = self.feature_value.get(&prop.name).unwrap();
+
+        let mut set = Default::default();
+        collect_string_alias_values(alias_type, def_type, def_value, &mut set);
+        set
+    }
+}
+
+/// Takes
+/// - a string-alias type, StringAlias("TeammateName") / TeamMateName
+/// - a type definition of a wider collection of teammates: e.g. Map<TeamMateName, TeamMate>
+/// - an a value for the collection of teammates: e.g. {"Alice": {}, "Bonnie": {}, "Charlie": {}, "Dawn"}
+///
+/// and fills a hash set with the full set of TeamMateNames, in this case: ["Alice", "Bonnie", "Charlie", "Dawn"]
+fn collect_string_alias_values(
+    alias_type: &TypeRef,
+    def_type: &TypeRef,
+    def_value: &Value,
+    set: &mut BTreeSet<String>,
+) {
+    match (def_type, def_value) {
+        (TypeRef::StringAlias(_), Value::String(s)) if alias_type == def_type => {
+            set.insert(s.clone());
+        }
+        (TypeRef::Option(dt), dv) if dv != &Value::Null => {
+            collect_string_alias_values(alias_type, dt, dv, set);
+        }
+        (TypeRef::EnumMap(kt, _), Value::Object(map)) if alias_type == &**kt => {
+            set.extend(map.keys().cloned());
+        }
+        (TypeRef::EnumMap(_, vt), Value::Object(map))
+        | (TypeRef::StringMap(vt), Value::Object(map)) => {
+            for item in map.values() {
+                collect_string_alias_values(alias_type, vt, item, set);
+            }
+        }
+        (TypeRef::List(vt), Value::Array(array)) => {
+            for item in array {
+                collect_string_alias_values(alias_type, vt, item, set);
+            }
+        }
+        _ => {}
+    }
+}
+
+#[cfg(test)]
+mod string_alias {
+
+    use super::*;
+    use serde_json::json;
+
+    fn test_set(alias_type: &TypeRef, def_type: &TypeRef, def_value: &Value, set: &[&str]) {
+        let mut observed = Default::default();
+        collect_string_alias_values(alias_type, def_type, def_value, &mut observed);
+
+        let expected: BTreeSet<_> = set.iter().map(|s| s.to_string()).collect();
```

Comment:
Always prefer `.to_owned()` to transmute a `&str` to a `String`. `to_string()` is implemented via `Display` traits and will invoke all the associated machinery there (which may take multiple allocations) and `.to_owned()` will just be `malloc()` + `memcpy()`

Reply:
Nice!